### PR TITLE
EmbeddedDeploymentManager

### PR DESCRIPTION
**Pull request by mproch**

### CODE REVIEW

File: engine/base/runtime/src/main/scala/pl/touk/nussknacker/engine/baseengine/metrics/dropwizard/DropwizardMetricsProviderForScenario.scala [hunk 1]
```diff
@@ -13,6 +14,12 @@ class DropwizardMetricsProviderFactory(metricRegistry: MetricRegistry) extends (
   override def apply(scenarioId: String): MetricsProviderForScenario with AutoCloseable = new DropwizardMetricsProviderForScenario(scenarioId, metricRegistry)
 }
 
+object DropwizardMetricsProviderForScenario {
+
+  val scenarioTagName = "process"
```
Comment (dswiecki) [inline engine/base/runtime/src/main/scala/pl/touk/nussknacker/engine/baseengine/metrics/dropwizard/DropwizardMetricsProviderForScenario.scala:19]: maybe we would change it to `scenario` tag ?

Comment (mproch) [reply engine/base/runtime/src/main/scala/pl/touk/nussknacker/engine/baseengine/metrics/dropwizard/DropwizardMetricsProviderForScenario.scala:19]: We'll do it in next PRs

File: build.sbt [hunk 2]
```diff
@@ -348,11 +348,13 @@ lazy val dist = {
         baseComponents / Compile / assembly,
         kafkaComponents / Compile / assembly,
         liteBaseComponents / Compile / assembly,
+        liteEmbeddedDeploymentManager / Compile /assembly
       ).value,
       Universal / mappings ++= Seq(
         (generic / crossTarget).value / "genericModel.jar" -> "model/genericModel.jar",
         (flinkDeploymentManager / crossTarget).value / "nussknacker-flink-manager.jar" -> "managers/nussknacker-flink-manager.jar",
         (engineStandalone / crossTarget).value / "nussknacker-standalone-manager.jar" -> "managers/nussknacker-standalone-manager.jar",
+        (liteEmbeddedDeploymentManager / crossTarget).value / "lite-embedded-deploymentManager.jar" -> "managers/lite-embedded-deploymentManager.jar",
```
Comment (dswiecki) [inline build.sbt:357]: lite-embedded-manager.jar ?

Comment (mproch) [reply build.sbt]: :+1:

File: engine/base/embeddedDeploymentManager/src/main/scala/pl/touk/nussknacker/streaming/embedded/EmbeddedDeploymentManager.scala [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        interpreters -= name
+        interpreter.close()
+        logger.debug(s"Scenario $name stopped")
+      }
+    }
+  }
+
+  override def findJobStatus(name: ProcessName): Future[Option[ProcessState]] = Future.successful {
+    interpreters.get(name).map { interpreterData =>
+      ProcessState(interpreterData.deploymentId, SimpleStateStatus.Running,
```
[the remark this answers is not in the document]
Comment (dswiecki) [reply engine/base/embeddedDeploymentManager/src/main/scala/pl/touk/nussknacker/streaming/embedded/EmbeddedDeploymentManager.scala:97]: resolved ;)

File: build.sbt [hunk 4]
```diff
@@ -884,7 +890,20 @@ lazy val liteModel = (project in engine("base/model")).
   settings(assemblySampleSettings("liteModel.jar"): _*).
   settings(
     name := "nussknacker-lite-model"
-  ).dependsOn(api % "provided", modelUtil % "provided")
+  ).dependsOn(api % "provided", modelUtil)
+
+lazy val liteEmbeddedDeploymentManager = (project in engine("base/embeddedDeploymentManager")).
+  configs(IntegrationTest).
+  settings(itSettings()).
+  enablePlugins().
```
[the remark this answers is not in the document]
Comment (mproch) [reply build.sbt:897]: copied by mistake I think, removed

File: build.sbt [hunk 5]
```diff
@@ -884,7 +890,20 @@ lazy val liteModel = (project in engine("base/model")).
...
+  settings(commonSettings).
+  settings(assemblySettings("lite-embedded-deploymentManager.jar", includeScala = false): _*).
```
[the remark this answers is not in the document]
Comment (mproch) [reply build.sbt]: :+1: change to better name

File: build.sbt (hunk 4 above)
[the remark this answers is not in the document]
Comment (mproch) [reply build.sbt:897]: copied by mistake, I removed

File: build.sbt [hunk 6]
```diff
@@ -884,7 +890,20 @@ lazy val liteModel = (project in engine("base/model")).
...
+
+  settings(
+    name := "lite-embedded-deploymentManager",
+    libraryDependencies ++= Seq(
```
[the remark this answers is not in the document]
Comment (mproch) [reply build.sbt]: removed

File: nussknacker-dist/src/universal/conf/dev-application.conf [hunk 7]
```diff
@@ -101,6 +105,33 @@ scenarioTypes {
...
+    modelConfig {
+      classPath: [ ${liteModelDir}/"liteModel.jar", ${openApiComponentDir}"/openapi.jar", ${sqlComponentDir}"/sql.jar" ]
+      categories: ["LiteStreaming"]
+      //FIXME: replace with final components
+      components.kafkaSources.enabled: true
+      kafka {
+        kafkaAddress: ${?KAFKA_ADDRESS}
+        kafkaProperties.retries: 1
+      }
+      exceptionHandlingConfig: {
```
[the remark this answers is not in the document]
Comment (mproch) [reply nussknacker-dist/src/universal/conf/dev-application.conf:129]: removed, added comment

**Review by dswiecki — APPROVED** [review]

**Pull request merged**